### PR TITLE
[release/6.0] Backport 'Fix Blazor RemoteAuthenticatorView processing the same action multiple times'

### LINKED ISSUE #43925: Backport the fix from #43844 to .NET 6 LTS
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The bug fix for the RemoteAuthenticatorView firing the LoginSucceeded callback multiple times has been fixed in Issue #43844. However, this code fix should be back-ported to .NET 6 since it is an LTS release and this issue affects so many people trying to use the LoginSucceeded callback to perform additional work.

### Expected Behavior

The LoginSucceeded callback should only ever fire once. In fact, ALL callbacks in RemoteAuthenticatorView should fire only once during any given authentication flow so that they are predictable and trustworthy.

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version

6.0.x

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by MackinnonBuck**
# [release/6.0] Backport 'Fix Blazor RemoteAuthenticatorView processing the same action multiple times'

Backport of #43844

Fixes an issue where the `<RemoteAuthenticatorView/>` component would sometimes fire the login callback more than once.

## Description

The previous logic processed the current authentication action each time the component was re-rendered. Depending on the authentication service used, this could result in the login callback firing more than once. This PR fixes the problem by caching the last processed action and disabling authentication processing until the action changes.

Fixes #43925

## Customer Impact

It causes difficulty for customers to write their own logic in a way that accounts for login callbacks firing more than once. In addition, other expensive authentication actions may get invoked unnecessarily behind the scenes, which can degrade the experience of the customer's app.

## Regression?

- [ ] Yes
- [X] No

## Risk

- [ ] High
- [ ] Medium
- [X] Low

The fix for this issue is simple, and we have extensive existing test coverage. New tests have been added validating the fix.

## Verification

- [X] Manual (required)
- [X] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [X] N/A

### CODE REVIEW

**Comment by MackinnonBuck** [comment]
I've also backported another fix that makes the components-e2e pipeline work on the latest versions of Chrome. This only affects the tests and is not a product change.

**Comment by mkArtakMSFT** [comment]
@dotnet/aspnet-build folks, should we wait for you to take care of this, or should I just merge this?

**Comment by wtgodbe** [comment]
One of us will merge it once the branches are open

**Comment by mkArtakMSFT** [comment]
> One of us will merge it once the branches are open

Thanks @wtgodbe.
We should automate this using FabricBot so that we don't ask this question every time there is an approved servicing PR. There should be a bot message saying "The PR is now approved. The @dotnet/aspnet-build team will merge it when branches open".
Let me know if you have concerns with this approach. If not, I'll just do this.

**Comment by wtgodbe** [comment]
> Let me know if you have concerns with this approach. If not, I'll just do this.

That sounds good!

**Comment by dougbu** [comment]
More generally, why does this question keep coming up❔ We consistently merge PRs that are green, approved, and have `servicing-approved` if they target servicing branches that are open. We also add comments in PRs targeting those branches, reminding owners to get them green or whatever's lacking. Is this just about reminding PR owners to wait patiently at other times❔

**Comment by mkArtakMSFT** [comment]
> More generally, why does this question keep coming up❔ We consistently merge PRs that are green, approved, and have `servicing-approved` if they target servicing branches that are open. We also add comments in PRs targeting those branches, reminding owners to get them green or whatever's lacking. Is this just about reminding PR owners to wait patiently at other times❔

Yes, this is just a reminder for PR owners to wait patiently.

**Comment by mkArtakMSFT** [comment]
Here is the PR to add the new FabricBot rule: https://github.com/dotnet/aspnetcore/pull/44735

**Review by mkArtakMSFT — APPROVED** [review]

**Pull request merged**
